File: DankestDungeon/Tile.java
package DankestDungeon;
public class Tile
{
   private A_Event event;
   
   public Tile()
   {
      event = new EmptyEvent();
   }
   public Tile(String type)
   {
      if(type.equals("w"))
      {
         event = new WallEvent();
      }
      if(type.equals("d"))
      {
         event = new DoorEvent();
      }
      if(type.equals("p"))
      {
         event = new PlayerEvent();
      }
   }
   
   public int occur()
   {
      return this.event.occur();
   }
   
   public String toString()
   {
      return event.toString();
   }
}

File: Tile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Chrisistheon
{
   public class Tile
   {
      private A_Event eve;
   
      public Tile()
      {
         eve = new EmptyEvent();
      }

      public Tile(string type)
      {
         if(type == "w")
         {
            eve = new WallEvent();
         }
         if(type == "d")
         {
            eve = new DoorEvent();
         }
         if(type == "p")
         {
            eve = new PlayerEvent();
         }
      }
   
      public int occur()
      {
         return this.eve.occur();
      }
   
      public string toString()
      {
         return eve.toString();
      }
   }
}
